### PR TITLE
Refactoring fio metrics

### PR DESCRIPTION
**Pull request by adata111**

### CODE REVIEW

File: perfmetrics/scripts/fio/fio_metrics.py [hunk 1]
```diff
@@ -43,12 +44,130 @@
...
+FORMAT = 'format'
+DEFAULT = 'default'
+LEVELS = 'levels'
+CONVERSION = 'conversion'
+NS_TO_S = 10**(-9)
+
+""" REQ_JOB_PARAMS:
+NAME: key for the parameter, to be used when creating parameter dict for each
+Job
+JSON_NAME: Key for parameter inside 'global options'/'job options' dictionary
```
Comment (vadlakondaswetha) [inline perfmetrics/scripts/fio/fio_metrics.py:60]: its not clearly getting conveyed that we can use any value for NAME but JSON_NAME has to match the fio job spec. please update the comment

File: perfmetrics/scripts/fio/fio_metrics.py [hunk 2]
```diff
@@ -43,12 +44,130 @@
...
+  Ex: For output json = {"global options": {"filesize":"50M"}, "jobs": [
+  "job options": {"rw": "read"}]}
+  JSON_NAME for filesize will be "filesize" and that for readwrite will be "rw"
+FORMAT: Function returning formatted parameter value. Needed to convert
+parameter to plottable values
+  Ex: 'filesize' is obtained as '50M', but we need to convert it to integer
+  showing size in kb in order to maintain uniformity
+DEFAULT: Default value for the parameter
+
+Ex: output json = {"global options": {"filesize":"50M"}}
```
Comment (vadlakondaswetha) [inline perfmetrics/scripts/fio/fio_metrics.py:70]: you mean fio output json?

File: perfmetrics/scripts/fio/fio_metrics.py [hunk 3]
```diff
@@ -43,12 +44,130 @@
...
+        NAME: RW,
+        JSON_NAME: RW,
+        FORMAT: lambda val: val,
+        DEFAULT: 'read'
+    }
+]
+
+""" REQ_JOB_METRICS:
+NAME: key for the metric, to be used when creating metric dict for each job
+LEVELS: Keys for the metric inside 'read'/'write' dictionary in each job
```
Comment (vadlakondaswetha) [inline perfmetrics/scripts/fio/fio_metrics.py:111]: same comment as above. levels should match the key in fio job spec

File: perfmetrics/scripts/fio/fio_metrics.py [hunk 4]
```diff
@@ -108,6 +225,15 @@ def _convert_value(value, conversion_dict, default_unit=''):
 
 
 def _get_rw(rw_value):
+  """Converting read/randread/write/randwrite to just read/write.
+
+  Args:
+    rw_value: str, 'read'/'randread'/'write'/'randwrite'
```
Comment (vadlakondaswetha) [inline perfmetrics/scripts/fio/fio_metrics.py:231]: what is the need for single quotes here. just say accepted values are read/randread/write/randwrite

File: perfmetrics/scripts/fio/fio_metrics.py [hunk 5]
```diff
@@ -152,75 +278,37 @@ def _load_file_dict(self, filepath) -> Dict[str, Any]:
...
-      to a particualar job:
-        jobname, filesize, number of threads, IOPS, Bandwidth and latency (min,
-        max and mean)
+  def _get_start_end_times(self, out_json, job_params) -> List[Tuple[int]]:
+    """Returns start and end times of each job as a list.
 
     Args:
-      fio_out: JSON object representing the fio output
+      out_json : FIO json output
+      job_params: List of dicts, Parameters of all jobs
```
Comment (vadlakondaswetha) [inline perfmetrics/scripts/fio/fio_metrics.py:286]: do you mean each dictionary contains parameter of a job.  please rephrase the comment, its not clear.

File: perfmetrics/scripts/fio/fio_metrics.py [hunk 6]
```diff
@@ -152,75 +278,37 @@ def _load_file_dict(self, filepath) -> Dict[str, Any]:
...
+      List of start and end time tuples, one tuple for each job
+      Ex: [(1653027014, 1653027084), (1653027084, 1653027155)]
 
     """
-
-    if not fio_out:
-      raise NoValuesError('No data in json object')
-
-    global_filesize = ''
+    rw_rev_list = [job_param[RW] for job_param in reversed(job_params)]
```
Comment (vadlakondaswetha) [inline perfmetrics/scripts/fio/fio_metrics.py:293]: why do you need reversed. can you add a comment here.

**Review by vadlakondaswetha — CHANGES_REQUESTED** [review]
please add test cases as well

File: perfmetrics/scripts/fio/fio_metrics.py (hunk 1 above)
Comment (adata111) [reply perfmetrics/scripts/fio/fio_metrics.py]: done

File: perfmetrics/scripts/fio/fio_metrics.py (hunk 2 above)
Comment (adata111) [reply perfmetrics/scripts/fio/fio_metrics.py]: done

File: perfmetrics/scripts/fio/fio_metrics.py (hunk 3 above)
Comment (adata111) [reply perfmetrics/scripts/fio/fio_metrics.py]: done

File: perfmetrics/scripts/fio/fio_metrics.py (hunk 5 above)
Comment (adata111) [reply perfmetrics/scripts/fio/fio_metrics.py]: done

File: perfmetrics/scripts/fio/fio_metrics.py (hunk 6 above)
Comment (adata111) [reply perfmetrics/scripts/fio/fio_metrics.py:330]: done

**Comment by adata111** [comment]
Added new tests and modified previous tests according to changes in fio_metrics.py

**Review by vadlakondaswetha — APPROVED** [review]

**Pull request merged**